light-zk.js/src/transaction/indexRecentTransactions.ts

```diff
@@ -308,58 +321,62 @@ const getTransactionsBatch = async ({
   );
 
   const lastSignature = signatures[signatures.length - 1];
-  let txs: (ParsedTransactionWithMeta | null)[] = [];
-  let index = 0;
   const signaturesPerRequest = 25;
 
-  while (index < signatures.length) {
-    try {
-      const txsBatch = await connection.getParsedTransactions(
-        signatures
-          .slice(index, index + signaturesPerRequest)
-          .map((sig) => sig.signature),
-        {
-          maxSupportedTransactionVersion: 0,
-          commitment: "confirmed",
-        },
-      );
-
-      if (!txsBatch.some((t) => !t)) {
-        txs = txs.concat(txsBatch);
-        index += signaturesPerRequest;
-      }
-    } catch (e) {
-      console.log("retry");
-      await sleep(2000);
-    }
-  }
+  let txs = [];
+  for (let i = 0; i < signatures.length; i += signaturesPerRequest) {
+    const sigsBatch = signatures
+      .slice(i, i + signaturesPerRequest)
+      .map((sig) => sig.signature);
 
-  const indexerEventTransactions = txs.filter((tx: any) => {
-    const accountKeys = tx.transaction.message.accountKeys;
-    const splNoopIndex = accountKeys.findIndex((item: ParsedMessageAccount) => {
-      const itemStr =
-        typeof item === "string" || item instanceof String
-          ? item
-          : item.pubkey.toBase58();
-      return itemStr === new PublicKey(SPL_NOOP_ADDRESS).toBase58();
-    });
-
-    if (splNoopIndex) {
-      return txs;
-    }
-  });
+    const txsBatch = await fetchTransactions(connection, sigsBatch);
+    txs.push(...txsBatch);
+  }
 
+  const filteredTxs = filterFetchedTransactions(txs);
   const indexerTransactionEvents = await processIndexerEventsTransactions(
-    indexerEventTransactions,
+    filteredTxs,
   );
 
   indexerTransactionEvents.forEach((event) => {
-    processIndexedTransaction(event!, transactions);
+    processIndexedTransaction(event, transactions);
   });
 
   return lastSignature;
 };
 
+const fetchTransactions = async (connection: Connection, sigs: string[]) => {
+  try {
+    const txsBatch = await connection.getParsedTransactions(sigs, {
+      maxSupportedTransactionVersion: 0,
+      commitment: "confirmed",
+    });
+
+    return txsBatch.every((t) => t) ? txsBatch : [];
+  } catch (e) {
+    console.log("retry");
```

Comment:
1. I think we should use `console.debug()` in library code, so we don't print stuff by default in someone's else's software. I'm fine with using `console.log()` in tests (or modules used only for testing), but not here.
2. We should make this message more descriptive - having just "retry" printed in the console wouldn't look too great. Can we include the info what are we retrying with a number of attempt?

Reply:
@vadorovsky makes totally sense. just extended the fetchTransactions method in order to support a proper retry logic by being more flexible in the setup. 🤙